pkg/cluster/manager/display.go

```diff
@@ -801,38 +789,52 @@ func (m *Manager) DisplayDashboardInfo(clusterName string, timeout time.Duration
 	}
 
 	ctx := context.WithValue(context.Background(), logprinter.ContextKeyLogger, m.logger)
-	pdAPI := api.NewPDClient(ctx, metadata.Topology.GetPDListWithManageHost(), timeout, tlsCfg)
-	dashboardAddr, err := pdAPI.GetDashboardAddress()
-	if err != nil {
-		return fmt.Errorf("failed to retrieve TiDB Dashboard instance from PD: %s", err)
-	}
-	if dashboardAddr == "auto" {
-		return fmt.Errorf("TiDB Dashboard is not initialized, please start PD and try again")
-	} else if dashboardAddr == "none" {
-		return fmt.Errorf("TiDB Dashboard is disabled")
+	return m.displayDashboards(ctx, metadata.Topology, nil, timeout, tlsCfg, metadata.Topology.GetPDListWithManageHost()...)
+}
+
+func (m *Manager) displayDashboards(ctx context.Context, t *spec.Specification, j *JSONOutput, timeout time.Duration, tlsCfg *tls.Config, pdList ...string) error {
+	dashboardAddrs := []string{}
+	t.IterInstance(func(ins spec.Instance) {
+		if ins.Role() != spec.ComponentDashboard {
+			return
+		}
+		dashboardAddrs = append(dashboardAddrs, utils.JoinHostPort(ins.GetManageHost(), ins.GetPort()))
+	})
+
+	pdDashboardAddr, err := t.GetPDDashboardAddress(ctx, tlsCfg, timeout, pdList...)
+	if err == nil && !set.NewStringSet("", "auto", "none").Exist(pdDashboardAddr) {
+		dashboardAddrs = append(dashboardAddrs, pdDashboardAddr)
 	}
 
-	u, err := url.Parse(dashboardAddr)
-	if err != nil {
-		return fmt.Errorf("unknown TiDB Dashboard PD instance: %s", dashboardAddr)
+	if len(dashboardAddrs) == 0 {
+		return fmt.Errorf("TiDB Dashboard is missing, try again later")
 	}
 
-	u.Path = "/dashboard/"
+	for i, addr := range dashboardAddrs {
+		scheme := "http"
 
-	if tlsCfg != nil {
-		fmt.Println(
-			"Client certificate:",
-			color.CyanString(m.specManager.Path(clusterName, spec.TLSCertKeyDir, spec.PFXClientCert)),
```

Comment:
I believe PFX format is deprecated, right @kaaaaaaang ?

Reply:
I don't know anything about its deprecation.

Reply:
Bring back the old code for now.